### PR TITLE
Spring Clean

### PR DESCRIPTION
**Pull request by joshghent**
Hi,

Can't remember the exact list of changes but this is basically a spring cleaning PR before I dive into adding some new features. Me and a few others want to get the codemirror version updated so we can scroll! :laughing: 

- Fixed codemirror not scrolling
- Fixed express not starting
- Updated all dev dependancies and dependancies
- Added guide to contributing
- Added setup guide to readme
- Fixed all jshint and jscs tests to use newest methods (a few rules had been deprecated)
- Updated travis node version to the latest.

Going forward I will be adding new features :smile: 

Thanks,

Josh

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 node_modules/
 .DS_Store
+.env
```
Comment (jmcriffey) [inline .gitignore]: Make sure files have new lines at the end.

File: .jscs.json [hunk 2]
```diff
@@ -8,10 +8,16 @@
         "beforeOpeningRoundBrace": true
     },
     "requireMultipleVarDecl": true,
-    "disallowLeftStickedOperators": ["?", "-", "+", "/", "*", "=", "==", "===", "!=", "!==", ">", ">=", "<", "<="],
-    "disallowRightStickedOperators": ["?", "/", "+", "*", ":", "=", "==", "===", "!=", "!==", ">", ">=", "<", "<="],
-    "requireRightStickedOperators": ["!"],
-    "requireLeftStickedOperators": [","],
+    "requireSpaceBeforeBinaryOperators": [
```
Comment (jmcriffey) [inline .jscs.json]: Thanks for updating this. I'm probably going to want to move on to eslint eventually.

Comment (joshghent) [inline .jscs.json]: Yeah, sounds good. I'd prefer that too. Perhaps we can transition to use a preset config? I have been using the airbnb one in my other projects but up to you :)

File: api.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ var mongo = require('mongodb').MongoClient,
     fiddles = null;
 
 module.exports = function(app) {
-    mongo.connect(String(process.env.MONGOHQ_URL), function(err, db) {
+    mongo.connect(process.env.MONGOHQ_URL, function(err, db) {
```
Comment (jmcriffey) [inline api.js]: Was this causing trouble somewhere? I can't remember why we were casting to a string here, so I am hesitant to remove it.

File: sample.env [hunk 4]
```diff
@@ -0,0 +1 @@
+MONGOHQ_URL='mongodb://localhost:27017/fiddles'
```
Comment (jmcriffey) [inline sample.env]: newline

**Comment by jmcriffey** [comment]
Looking good. Just had a few suggestions and questions. Thanks!

**Comment by joshghent** [comment]
I've made the changes based on the suggestions you've made. Should be good to merge now :)

**Pull request merged**